### PR TITLE
Bratseth/simplify validate zones

### PR DESCRIPTION
**Pull request by bratseth**
@mpolden how about this?

### CODE REVIEW

**Comment by bratseth** [comment]
Please ignore the first commit. I accidentally branched off from that ...

File: config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecTest.java [hunk 1]
```diff
@@ -276,8 +276,7 @@ public void productionSpecWithDuplicateRegions() {
             DeploymentSpec.fromXml(r);
             fail("Expected exception");
         } catch (IllegalArgumentException e) {
-            assertEquals("All declared regions must be unique, but found these duplicated regions: " +
-                                 "[us-west-1, us-central-1]", e.getMessage());
+            assertEquals("prod.us-west-1 is listed twice in deployment.xml", e.getMessage());
```
Comment (mpolden) [inline config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecTest.java]: Seems it no longer lists all duplicates. It fails to mention "prod.us-central-1" which appears in both `<parallel>` tags.

Comment (bratseth) [reply config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecTest.java]: Sure, but I don't think it's ever worth additional code to create more complete error messages like that: very small probability * very small added value

File: config-model-api/src/main/java/com/yahoo/config/application/api/DeploymentSpec.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ private static void validateTotalDelay(List<Step> steps) {
     }
 
     /** Throw an IllegalArgumentException if any production zone is declared multiple times */
-    private static void validateZones(List<Step> steps) {
+    private static void validateZonesOld(List<Step> steps) {
```
Comment (mpolden) [inline config-model-api/src/main/java/com/yahoo/config/application/api/DeploymentSpec.java]: Remove method?

Comment (bratseth) [reply config-model-api/src/main/java/com/yahoo/config/application/api/DeploymentSpec.java]: Yes, sorry about that.

File: config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecTest.java (hunk 1 above)
Comment (mpolden) [reply config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecTest.java]: Ok, test input should be simplified as well then, second `<parallel>` tag can be removed.

Comment (bratseth) [reply config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecTest.java]: ok, done

**Comment by mpolden** [comment]
LGTM.

**Pull request merged**